### PR TITLE
sets: Rename kinds with a more consistent naming scheme.

### PR DESCRIPTION
**Pull request by aniemetz**
This prefixes sets kinds with `SET_` and relation kinds with
`RELATION_`. It also prefixes the corresponding SMT-LIB operators with
`set.` and relation operators with `rel.`.

### CODE REVIEW

File: test/regress/regress1/fmf/nlp042+1.smt2 [hunk 1]
```diff
@@ -64,8 +64,8 @@
 (declare-fun nonexistent ($$unsorted $$unsorted) Bool)
 (assert (forall ((U $$unsorted) (V $$unsorted)) (=> (eventuality U V) (nonexistent U V)) ))
 (assert (forall ((U $$unsorted) (V $$unsorted)) (=> (eventuality U V) (specific U V)) ))
-(declare-fun singleton ($$unsorted $$unsorted) Bool)
-(assert (forall ((U $$unsorted) (V $$unsorted)) (=> (thing U V) (singleton U V)) ))
+(declare-fun set.singleton ($$unsorted $$unsorted) Bool)
```
Comment (ajreynol) [inline test/regress/regress1/fmf/nlp042+1.smt2:67]: This should not change.

File: test/regress/regress1/fmf/agree466.smt2 [hunk 2]
```diff
@@ -258,10 +258,10 @@
 (assert (forall ((mk_array_7_index node$type)) (= (select mk_array_7
                                                   mk_array_7_index) null_state)))
 (define-fun null_state_set () state_set$type mk_array_7)
-(declare-fun choose (Values$t$type) value$type)
+(declare-fun set.choose (Values$t$type) value$type)
```
Comment (ajreynol) [inline test/regress/regress1/fmf/agree466.smt2:261]: This should not change (the whole benchmark).

File: test/regress/regress0/tptp/SYN000-2.p [hunk 3]
```diff
@@ -18,7 +18,7 @@
 %            Maximal clause size   :    7 (   1 average)
 %            Number of predicates  :    8 (   3 propositional; 0-3 arity)
 %            Number of functors    :   22 (  20 constant; 0-3 arity)
-%            Number of variables   :    7 (   7 singleton)
+%            Number of variables   :    7 (   7 set.singleton)
```
Comment (ajreynol) [inline test/regress/regress0/tptp/SYN000-2.p:21]: The TPTP benchmarks should not change in this PR.

File: test/regress/regress1/fmf/nlp042+1.smt2 (hunk 1 above)
Comment (aniemetz) [reply test/regress/regress1/fmf/nlp042+1.smt2:67]: good catch

File: src/theory/bags/theory_bags_type_rules.cpp [hunk 4]
```diff
@@ -239,7 +239,7 @@ TypeNode ChooseTypeRule::computeType(NodeManager* nodeManager,
     if (!bagType.isBag())
     {
       throw TypeCheckingExceptionPrivate(
-          n, "CHOOSE operator expects a bag, a non-bag is found");
+          n, "SET_CHOOSE operator expects a bag, a non-bag is found");
```
Comment (mudathirmahgoub) [inline src/theory/bags/theory_bags_type_rules.cpp:242]:
```suggestion
          n, "BAG_CHOOSE operator expects a bag, a non-bag is found");
```

File: src/api/cpp/cvc5.cpp [hunk 5]
```diff
@@ -273,26 +273,27 @@ const static std::unordered_map<Kind, cvc5::Kind> s_kinds{
...
+    {SET_IS_SINGLETON, cvc5::Kind::SET_IS_SINGLETON},
+    /* Relations ----------------------------------------------------------- */
+    {RELATION_JOIN, cvc5::Kind::RELATION_JOIN},
+    {RELATION_PRODUCT, cvc5::Kind::RELATION_PRODUCT},
+    {RELATION_TRANSPOSE, cvc5::Kind::RELATION_TRANSPOSE},
+    {RELATION_TCLOSURE, cvc5::Kind::RELATION_TCLOSURE},
+    {RELATION_JOIN_IMAGE, cvc5::Kind::RELATION_JOIN_IMAGE},
+    {RELATION_IDEN, cvc5::Kind::RELATION_IDEN},
     /* Bags ---------------------------------------------------------------- */
     {UNION_MAX, cvc5::Kind::UNION_MAX},
```
Comment (mudathirmahgoub) [inline src/api/cpp/cvc5.cpp:298]:
```suggestion
    {BAG_UNION_MAX, cvc5::Kind::BAG_UNION_MAX},
```

File: src/api/cpp/cvc5.cpp [hunk 6]
```diff
@@ -273,26 +273,27 @@ const static std::unordered_map<Kind, cvc5::Kind> s_kinds{
...
     {UNION_DISJOINT, cvc5::Kind::UNION_DISJOINT},
```
Comment (mudathirmahgoub) [inline src/api/cpp/cvc5.cpp:299]:
```suggestion
    {BAG_UNION_DISJOINT, cvc5::Kind::BAG_UNION_DISJOINT},
```

Comment (mudathirmahgoub) [reply src/api/cpp/cvc5.cpp:299]: for the others as well
`BAG_INTERSECTION_MIN`, `BAG_DIFFERENCE_SUBTRACT`, `BAG_DIFFERENCE_REMOVE`, `BAG_SUBBAG`
`BAG_DUPLICATE_REMOVAL`, `BAG_EMPTY`. 
Not sure what the appropriate name for the bag constructor. Perhaps  `BAG_MK`?

Comment (mudathirmahgoub) [reply src/api/cpp/cvc5.cpp:299]: I just realized that the PR is only about sets, so I can handle bag refactoring in another PR.

File: src/printer/smt2/smt2_printer.cpp [hunk 7]
```diff
@@ -330,7 +330,7 @@ void Smt2Printer::toStream(std::ostream& out,
       out << ss.str();
       break;
     }
-    case kind::EMPTYSET:
+    case kind::SET_EMPTY:
       out << "(as emptyset ";
```
Comment (mudathirmahgoub) [inline src/printer/smt2/smt2_printer.cpp:334]:
```suggestion
      out << "(as set.empty ";
```

File: src/printer/smt2/smt2_printer.cpp [hunk 8]
```diff
@@ -698,7 +698,7 @@ void Smt2Printer::toStream(std::ostream& out,
     return;
   }
   break;
-  case kind::UNIVERSE_SET:out << "(as univset " << n.getType() << ")";break;
+  case kind::SET_UNIVERSE: out << "(as univset " << n.getType() << ")"; break;
```
Comment (mudathirmahgoub) [inline src/printer/smt2/smt2_printer.cpp:701]:
```suggestion
  case kind::SET_UNIVERSE: out << "(as set.universe " << n.getType() << ")"; break;
```

File: src/proof/lfsc/lfsc_node_converter.cpp [hunk 9]
```diff
@@ -306,15 +306,15 @@ Node LfscNodeConverter::postConvert(Node n)
...
+  else if (k == SET_EMPTY || k == SET_UNIVERSE || k == EMPTYBAG)
   {
     Node t = typeAsNode(convertType(tn));
     TypeNode etype = nm->mkFunctionType(d_sortType, tn);
     Node ef = getSymbolInternal(
         k,
         etype,
-        k == EMPTYSET ? "emptyset"
-                      : (k == UNIVERSE_SET ? "univset" : "emptybag"));
+        k == SET_EMPTY ? "emptyset"
```
Comment (mudathirmahgoub) [inline src/proof/lfsc/lfsc_node_converter.cpp:316]:
```suggestion
        k == SET_EMPTY ? "set.empty"
```

File: src/proof/lfsc/lfsc_node_converter.cpp [hunk 10]
```diff
@@ -306,15 +306,15 @@ Node LfscNodeConverter::postConvert(Node n)
...
+                       : (k == SET_UNIVERSE ? "univset" : "emptybag"));
```
Comment (mudathirmahgoub) [inline src/proof/lfsc/lfsc_node_converter.cpp:317]:
```suggestion
                       : (k == SET_UNIVERSE ? "set.universe" : "emptybag"));
```

File: src/theory/bags/bags_rewriter.cpp [hunk 11]
```diff
@@ -461,7 +461,7 @@ BagsRewriteResponse BagsRewriter::rewriteIsSingleton(const TNode& n) const
 BagsRewriteResponse BagsRewriter::rewriteFromSet(const TNode& n) const
 {
   Assert(n.getKind() == BAG_FROM_SET);
-  if (n[0].getKind() == SINGLETON)
+  if (n[0].getKind() == SET_SINGLETON)
   {
     // (bag.from_set (singleton (singleton_op Int) x)) = (mkBag x 1)
```
Comment (mudathirmahgoub) [inline src/theory/bags/bags_rewriter.cpp:466]:
```suggestion
    // (bag.from_set (set.singleton (singleton_op Int) x)) = (mkBag x 1)
```

File: src/theory/sets/theory_sets_rels.cpp [hunk 12]
```diff
@@ -269,11 +282,13 @@ void TheorySetsRels::check(Theory::Effort level)
     Trace("rels-debug") << "[Theory::Rels] Done with collecting relational terms!" << std::endl;
   }
 
-  /* JOIN-IMAGE UP  :   (x, x1) IS_IN R, ..., (x, xn) IS_IN R  (R JOIN_IMAGE n)
-  *                     -------------------------------------------------------
-  *                     x IS_IN (R JOIN_IMAGE n) || NOT DISTINCT(x1, ... , xn)
-  *
-  */
+  /* RELATION_JOIN-IMAGE UP  :   (x, x1) IS_IN R, ..., (x, xn) IS_IN R  (R
```
Comment (mudathirmahgoub) [inline src/theory/sets/theory_sets_rels.cpp:285]: Please can you format this inference rule?

File: src/theory/sets/theory_sets_rels.cpp [hunk 13]
```diff
@@ -365,7 +380,7 @@ void TheorySetsRels::check(Theory::Effort level)
     Trace("rels-debug") << "\n[Theory::Rels] *********** Done with computing members for JoinImage Term" << join_image_term << "*********** " << std::endl;
   }
 
-  /* JOIN-IMAGE DOWN  : (x) IS_IN (R JOIN_IMAGE n)
+  /* RELATION_JOIN-IMAGE DOWN  : (x) IS_IN (R RELATION_JOIN_IMAGE n)
```
Comment (mudathirmahgoub) [inline src/theory/sets/theory_sets_rels.cpp:383]: ditto

File: src/theory/strings/regexp_operation.cpp [hunk 14]
```diff
@@ -1422,7 +1422,7 @@ Node RegExpOpr::removeIntersection(Node r) {
   Assert(!visited.find(r)->second.isNull());
   if (Trace.isOn("regexp-intersect"))
   {
-    Trace("regexp-intersect") << "Remove INTERSECTION( " << mkString(r)
+    Trace("regexp-intersect") << "Remove SET_INTERSECTION( " << mkString(r)
```
Comment (mudathirmahgoub) [inline src/theory/strings/regexp_operation.cpp:1425]:
```suggestion
    Trace("regexp-intersect") << "Remove INTERSECTION( " << mkString(r)
```

File: src/theory/strings/regexp_operation.cpp [hunk 15]
```diff
@@ -1439,11 +1439,11 @@ Node RegExpOpr::intersect(Node r1, Node r2)
   std::map<PairNodes, Node> cache;
   Trace("regexp-intersect-node") << "Intersect (1): " << rr1 << std::endl;
   Trace("regexp-intersect-node") << "Intersect (2): " << rr2 << std::endl;
-  Trace("regexp-intersect") << "Start INTERSECTION(\n\t" << mkString(r1)
+  Trace("regexp-intersect") << "Start SET_INTERSECTION(\n\t" << mkString(r1)
```
Comment (mudathirmahgoub) [inline src/theory/strings/regexp_operation.cpp:1442]:
```suggestion
  Trace("regexp-intersect") << "Start INTERSECTION(\n\t" << mkString(r1)
```

File: src/theory/strings/regexp_operation.cpp [hunk 16]
```diff
@@ -1439,11 +1439,11 @@ Node RegExpOpr::intersect(Node r1, Node r2)
...
                             << ",\n\t" << mkString(r2) << ")" << std::endl;
   Node retNode = intersectInternal(rr1, rr2, cache, 1);
   Trace("regexp-intersect")
-      << "End INTERSECTION(\n\t" << mkString(r1) << ",\n\t" << mkString(r2)
+      << "End SET_INTERSECTION(\n\t" << mkString(r1) << ",\n\t" << mkString(r2)
```
Comment (mudathirmahgoub) [inline src/theory/strings/regexp_operation.cpp:1446]:
```suggestion
      << "End INTERSECTION(\n\t" << mkString(r1) << ",\n\t" << mkString(r2)
```

File: src/api/cpp/cvc5.cpp (hunk 5 above)
Comment (aniemetz) [reply src/api/cpp/cvc5.cpp:298]: We definitely want to refactor bags, but not in this PR.

File: src/api/cpp/cvc5.cpp (hunk 6 above)
Comment (aniemetz) [reply src/api/cpp/cvc5.cpp:299]: That would be great!

**Review by mudathirmahgoub — APPROVED** [review]
LGTM, although I skipped regression files which pass CI.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**